parachains/runtimes/collectives/collectives-polkadot/src/fellowship/mod.rs

```diff
@@ -0,0 +1,151 @@
+	pub const DAN_5: Rank = 5; // aka Experts.
+	pub const DAN_6: Rank = 6;
+	pub const DAN_7: Rank = 7; // aka Masters.
+	pub const DAN_8: Rank = 8;
+	pub const DAN_9: Rank = 9;
+}
+
+parameter_types! {
+	pub const AlarmInterval: BlockNumber = 1;
+	pub const SubmissionDeposit: Balance = 0;
+	pub const UndecidingTimeout: BlockNumber = 7 * DAYS;
+	pub const TechnicalCommittee: BodyId = BodyId::Technical;
+	// Referenda pallet account, used to temporarily deposit slashed imbalance before teleporting.
+	pub ReferendaPalletAccId: AccountId = constants::account::REFERENDA_PALLET_ID.into_account_truncating();
+	pub RelayTreasuryAccId: AccountId = constants::account::RELAY_TREASURY_PALL_ID.into_account_truncating();
+	pub const FellowshipAdminBodyId: BodyId = BodyId::Index(FELLOWSHIP_ADMIN_INDEX);
+}
+
+impl pallet_fellowship_origins::Config for Runtime {}
+
+pub type FellowshipReferendaInstance = pallet_referenda::Instance1;
+
+impl pallet_referenda::Config<FellowshipReferendaInstance> for Runtime {
+	type WeightInfo = weights::pallet_referenda::WeightInfo<Runtime>;
+	type RuntimeCall = RuntimeCall;
+	type RuntimeEvent = RuntimeEvent;
+	type Scheduler = Scheduler;
+	type Currency = Balances;
+	type SubmitOrigin =
+		pallet_ranked_collective::EnsureMember<Runtime, FellowshipCollectiveInstance, 1>;
+	type CancelOrigin = FellowshipExperts;
+	type KillOrigin = FellowshipMasters;
+	type Slash = ToParentTreasury<RelayTreasuryAccId, ReferendaPalletAccId, Runtime>;
+	type Votes = pallet_ranked_collective::Votes;
+	type Tally = pallet_ranked_collective::TallyOf<Runtime, FellowshipCollectiveInstance>;
+	type SubmissionDeposit = SubmissionDeposit;
+	type MaxQueued = ConstU32<100>;
+	type UndecidingTimeout = UndecidingTimeout;
+	type AlarmInterval = AlarmInterval;
+	type Tracks = tracks::TracksInfo;
+	type Preimages = Preimage;
+}
+
+pub type FellowshipCollectiveInstance = pallet_ranked_collective::Instance1;
+
+morph_types! {
+	/// A `TryMorph` implementation to reduce a scalar by a particular amount, checking for
+	/// underflow.
+	pub type CheckedReduceBy<N: TypedGet>: TryMorph = |r: N::Type| -> Result<N::Type, ()> {
+		r.checked_sub(&N::get()).ok_or(())
+	} where N::Type: CheckedSub;
+}
+
+// IsMajorityOfBody<KsmLocation, ExecutiveBody>,
+
+impl pallet_ranked_collective::Config<FellowshipCollectiveInstance> for Runtime {
+	type WeightInfo = weights::pallet_ranked_collective::WeightInfo<Runtime>;
+	type RuntimeEvent = RuntimeEvent;
+	// Promotion is by any of:
+	// - Root can promote arbitrarily.
+	// - a majority of the Technical Committee (for initialization, to be later removed);
+	// - a vote by the rank *above* the new rank.
+	type PromoteOrigin = EitherOf<
+		EitherOf<
+			frame_system::EnsureRootWithSuccess<Self::AccountId, ConstU16<65535>>,
+			MapSuccess<
+				EnsureXcm<IsMajorityOfBody<DotLocation, TechnicalCommittee>>,
+				Replace<ConstU16<6>>,
+			>,
+		>,
+		EitherOf<
+			MapSuccess<
+				EnsureXcm<IsVoiceOfBody<DotLocation, FellowshipAdminBodyId>>,
```

Comment:
`DotLocation` is a misuse here. It means "the location of the DOT token", which happens to be the same as "the location of the governance of the Relay-chain" for the moment, but when we have a separate governance chain, this will break. There should be a separate `RelayLocation` or better, `GovernanceLocation`, type-value instead.